Update "Using HTML sections and outlines"

mfuji09 opened the pull request:
https://github.com/mdn/content/issues/1303

Both the living standard and the W3C standard don't say "should be" nor "shouldn't be" for `<nav>`'s usage. So I've modified it so that it doesn't use these expressions for things that are not prohibited by the Living Standard.

mfuji09 committed:
Update "Using HTML sections and outlines"

https://github.com/mdn/content/issues/1303

Both the living standard and the W3C standard don't say "should be" nor "shouldn't be" for `<nav>`'s usage. So I've modified it so that it doesn't use these expressions for things that are not prohibited by the Living Standard.

chrisdavidmills commented on files/en-us/web/guide/html/using_html_sections_and_outlines/index.html:
```diff
@@ -95,7 +95,7 @@ <h2 id="How_Do_Sectional_Elements_Work">How Do Sectional Elements Work?</h2>
 <h3 id="Nav_Element">Nav Element</h3>
 
-<p>The <code>&lt;nav&gt;</code> element indicates a navigation block and should be used for major navigational menus.</p>
+<p>The <code>&lt;nav&gt;</code> element indicates a navigation block and is mainly used for sections that consists of major navigation blocks.</p>
```
"consists" -> "consist"

mfuji09 replied:
Thanks! I have fixed it.

chrisdavidmills commented on files/en-us/web/guide/html/using_html_sections_and_outlines/index.html:
```diff
@@ -124,7 +124,7 @@ <h4 id="Nesting_Menus">Nesting Menus</h4>
 <h4 id="Collection_of_Links">Collection of Links</h4>
 
-<p>Use the <code>&lt;nav&gt;</code> element only for site navigation menus. Collections of links such as social media profiles or a blogroll should not be wrapped in the <code>&lt;nav&gt;</code> element.</p>
+<p>The <code>&lt;nav&gt;</code> element is intended to be used to mark a section with navigation links. Not all collections of links on a page need to be in a <code>&lt;nav&gt;</code> element.</p>
```
I'm not convinced that the rewrite is better than the original here. One key point missing in my opinion is that `<nav>` is for marking up the page's *primary* navigation (whether that is a menu, or even a search box). That should be mentioned.

And I'd still like to see examples of non-primary navigation that doesn't go in a `<nav>` talked about, although "blogroll" is an out-of-date concept these days. Maybe talk about "social media profiles or related links"

Also perhaps mention that for non-primary navigation links, an `<aside>` is possibly a better element? Or mayb just a `<section>`?

mfuji09 replied:
@chrisdavidmills Thank you for comment. 
I have tried to rewrite it to show more specific usage. Can you re-review it?

chrisdavidmills reviewed (changes requested):
Some review comments for you to consider. Thanks @mfuji09 !

mfuji09 committed:
Update: Using html section and outlines

- fixed grammar
- "Collection of Links": more specific usage of `<nav>`, and other sectioning elements.

chrisdavidmills committed:
Update index.html

chrisdavidmills reviewed (approved):
I've had a go at rewriting the descriptions a bit more. Do you think these read OK now?
